app/controllers/synapses_controller.rb

```diff
@@ -22,10 +22,15 @@ def create
     @synapse = Synapse.new(synapse_params)
     @synapse.desc = '' if @synapse.desc.nil?
     @synapse.desc.strip! # no trailing/leading whitespace
-    authorize @synapse
+
+    success = nil
+    ActiveRecord::Base.transaction do
+      success = @synapse.save
+      success ? authorize(@synapse) : skip_authorization
```

Comment:
Now that I read this line on Github I should probably make it clearer. Originally I had it as

```
authorize(@synapse) if success
skip_authorization if !success
```